dwave/plugins/sklearn/nearest_neighbors.pyx

```diff
@@ -0,0 +1,117 @@
+# distutils: language = c++
+import numpy as np
+cimport numpy as np
+from libcpp.vector cimport vector
+
+cdef extern from "math.h":
+    double log(double x) nogil
+
+def calculate_mi(np.ndarray[np.float_t,ndim = 2] X,
+                np.ndarray[np.float_t,ndim = 1] y,
+                unsigned long refinement_factor = 5): 
+        
+    cdef unsigned long n_obs = X.shape[0]
+    cdef float log_n_obs = log(n_obs)
+
+    cdef unsigned long n_variables = X.shape[1]
+
+    if(n_obs != y.shape[0]):
+        raise ValueError("y is the wrong shape")
+
+    cdef long sub_index_size = refinement_factor*int(np.round(log(n_obs))) + 2
+    cdef total_state_space_size = pow(sub_index_size,3)
```

Comment:
I think you want [pow](https://github.com/cython/cython/blob/a174dcd02f1f662b61f66953929709fbb47b605e/Cython/Includes/libc/math.pxd#L156)?